ld_eventsource/testing/test_sse_client_retry.py

```diff
@@ -101,6 +102,57 @@ def test_all_iterator_continues_after_retry():
         assert client.next_retry_delay == initial_delay * 2
 
 
+def test_retry_delay_gets_reset_after_threshold():
+    initial_delay = 0.005
+    retry_delay_reset_threshold = 0.1
+    mock = MockConnectStrategy(
+        RespondWithData("data: data1\n\n"),
+        RejectConnection(HTTPStatusError(503)),
+    )
+    with SSEClient(
+        connect=mock,
+        error_strategy=ErrorStrategy.always_continue(),
+        initial_retry_delay=initial_delay,
+        retry_delay_reset_threshold=retry_delay_reset_threshold,
+        retry_delay_strategy=RetryDelayStrategy.default(jitter_multiplier=None),
+    ) as client:
+        assert client._retry_reset_baseline == 0
+        all = client.all
+
+        # Establish a successful connection
+        item1 = next(all)
+        assert isinstance(item1, Start)
+        assert client._retry_reset_baseline != 0
+
+        item2 = next(all)
+        assert isinstance(item2, Event)
+        assert item2.data == 'data1'
+
+        # Stream is dropped and then fails to re-connect, resulting in backoff.
+        item3 = next(all)
+        assert isinstance(item3, Fault)
+        assert client.next_retry_delay == initial_delay
+
+        item4 = next(all)
+        assert isinstance(item4, Fault)
+        assert client.next_retry_delay == initial_delay * 2
+
+        # Sleeping the threshold should reset the retry thresholds
+        sleep(retry_delay_reset_threshold)
```

Comment:
Maybe a little scary, but I assume we don't have an ability to mock advance time.